src/planscape/planning/views.py

```diff
@@ -164,38 +196,51 @@ def delete_planning_area(request: HttpRequest) -> HttpResponse:
         # Check that the user is logged in.
         user = request.user
         if not user.is_authenticated:
-            return JsonResponse({"error": "Authentication Required"}, status=401)
+            return Response(
+                {"error": "Authentication Required"},
+                status=status.HTTP_401_UNAUTHORIZED,
+            )
 
         # Get the planning area IDs
         body = json.loads(request.body)
         planning_area_id = body.get("id", None)
         planning_area_ids = []
         if planning_area_id is None:
-            raise ValueError("Must specify planning area id.")
+            return Response(
+                {"message": "No ID given for planning area"},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
         elif isinstance(planning_area_id, int):
             planning_area_ids = [planning_area_id]
         elif isinstance(planning_area_id, list):
             planning_area_ids = planning_area_id
         else:
-            raise ValueError("Planning Area ID must be an int or a list of ints.")
-
-        # Get the planning area(s) for just the logged in user.
-        planning_areas = user.planning_areas.filter(pk__in=planning_area_ids)
+            return Response(
+                {"message": "Planning Area ID must be an int or a list of ints."},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
 
-        planning_areas.delete()
+        planning_areas = PlanningArea.objects.get_for_user(user).filter(
+            pk__in=planning_area_ids
+        )
+        for p in planning_areas:
+            if PlanningAreaPermission.can_remove(user, p):
+                p.delete()
+            else:
+                logger.error(f"User {user} has no permission to delete {p.id}")
```

Comment:
Our existing endpoint accepts a list of planning areas, but it doesn't complain if someone tries to delete a planning area that either doesn't exist or if they don't have to access to it. I'm keeping that functionality for simplicity, but I'm logging that it happens.

I don't know if that's what we ultimately want, but one alternative migh be to create a list of failed deletions and return those in the response.